cluster/kube/client.go

```diff
@@ -98,6 +100,7 @@ func NewClient(ctx context.Context, log log.Logger, ns string, configPath string
 		ns:                ns,
 		log:               log.With("client", "kube"),
 		kubeContentConfig: config,
+		env:               util.EnvironmentVariablesToMap(),
```

Comment:
env variables must be validated for values and then read into dedicated struct fields

Reply:
Here is a direct conversion from the environment variables to a `map[string]string`. I can make validations but after this call. To have more fine-grained error handling on unexpected values.

Reply:
we would rather want to see dedicated struct fields instead of env converted to map. map is useful as intermediate store in some circumstances (like converting data to json). `Go` is a strict type of language and we use as much of it.

client instantiation is the last step where all configuration values must be validated

Reply:
It means we know the context in which the usage gave an error. Having a struct makes absolute sense! I'll change the code to support such configuration rather than environment variables

Reply:
We treat an error as an error, it is really hard to chase configuration errors on the fly.
1. the client must not interact with env variables, configs etc, the only way for it to receive configuration is thru a parameter to [NewClient](https://github.com/akash-network/provider/blob/718c95477f7be957474665a70391546cacf3185d/cluster/kube/client.go#L67).
2. Each environment variable must have a corresponding cli flag for it. That's why using env within the client breaks it.
3. Using environment variables inside implementation complicates writing unit tests.

remove environment variables from the client completely and use struct fields as values during annotations init.
I'm still unsure if using two different vars is the correct way to do it. if the workload going to use ns issuer then it has nothing to do with the cluster